src/Core/src/Handlers/Window/WindowHandler.Android.cs

```diff
@@ -88,13 +89,13 @@ internal static void DisconnectHandler(NavigationRootManager? navigationRootMana
 			navigationRootManager?.Disconnect();
 		}
 
-		internal static View? CreateRootViewFromContent(IWindowHandler handler, IWindow window)
+		internal static NavigationRootManager CreateRootViewFromContent(IWindowHandler handler, IWindow window)
 		{
 			_ = handler.MauiContext ?? throw new InvalidOperationException($"{nameof(MauiContext)} should have been set by base class.");
 
 			var rootManager = handler.MauiContext.GetNavigationRootManager();
 			rootManager.Connect(window.Content);
-			return rootManager.RootView;
+			return rootManager;
```

Comment:
```suggestion
			handler.PlatformView.SetContentView(rootManager.RootView);
			rootManager.PerformPendingFragmentTransaction();
```

Reply:
We could even go further and make the Connect method take an action or handler that will set the platform view so we don't have to store the pending transaction. But that may be overkill. The very least we can reduce the steps from 3 places to 2, making this easier.